src/libstd/ffi/c_str.rs

```diff
@@ -195,6 +195,7 @@ pub struct CString {
 /// [`from_ptr`]: #method.from_ptr
 #[derive(Hash)]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[repr(transparent)]
 pub struct CStr {
```

Comment:
Note that by adding this attribute you are making a semver commitment that it will stay this way forever. In this particular case, we even have a FIXME that says we want to change the representation to something else so it seems ill-advised to make such a commitment.